include/mbedtls/ssl.h

```diff
@@ -1200,7 +1200,12 @@ struct mbedtls_ssl_session
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3)
     mbedtls_ssl_tls13_application_secrets MBEDTLS_PRIVATE(app_secrets);
-#endif
+
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && \
```

Comment:
This could be moved in the "#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)" section above and the MBEDTLS_SSL_CLI_C guard must be added:
```C
#if defined( MBEDTLS_SSL_SERVER_NAME_INDICATION ) && \
    defined( MBEDTLS_SSL_CLI_C )
    char *MBEDTLS_PRIVATE(hostname); /*!< host name binded with tickets */
#endif
```